### PR TITLE
Add custom tokens

### LINKED ISSUE #676: Allow adding custom tokens to tunnel
### Feature or change description

Users should be able to tunnel tokens not listed on hemi.

See [Design](https://www.figma.com/design/4fVd9wneclsvYDYD95ApZ9/Hemi-Portal?node-id=5597-10634&node-type=frame&m=dev) here

### Proposed implementation

- Users will save the added token in their local storage
- Information of the token will be read from the L1/L2 chain, or added by the user if needed

### PR DESCRIPTION
**Pull request by gndelia**
### Description

This PR allow users to add custom tokens. For L1 tokens, the user needs to enter both the L1 and the L2 addresses. They are validated against each other.   
For L2 tokens, only the L2 address is needed, as the L1 is picked automatically. If it can't be loaded, it means the token is invalid, as we run the same checks as OP does to verify if the pair is supported by the bridges.

As of this PR, the token list will now be loaded from a hook on runtime, instead of being static. This is because it merges the static list (From the [@hemilabs/token-list](https://github.com/hemilabs/token-list)) with the locally saved list the user has. Note that the locally saved list is per-computer and not per address.

If a user enters their past history in another computer, the data of the token is loaded dynamically. However, to tunnel them again, they will need to add them again.

**Note:** In order to test this, I used `USDT` in testnet and `COMP` in mainnet, by locally removing them from the static token-list

Originally I intended to split this into 2 PRs but as it became really complex, it ended up in one large one. This was because adding custom tokens would break past operations if I did not merge all the changes. My bad!

Issues related:

https://github.com/hemilabs/ui-monorepo/issues/688
https://github.com/hemilabs/ui-monorepo/issues/689

### Screenshots

Adding a L1 custom token

https://github.com/user-attachments/assets/b6f3bdef-b0b7-4c00-913f-751783dad49b

Adding a L2 custom token (also showing an invalid address and a known token address). See how the L1 token is automatically loaded

https://github.com/user-attachments/assets/ac18abbe-266a-4089-ab26-949b7f1accf7

Tunnel operations will appear in the Tx history, regardless of the token being added or not in the custom list

image

Some mobile screenshots

image

image

### Related issue(s)

Closes #676

### Checklist

- [x] Manual testing passed.
- [ ] Automated tests added, or N/A.
- [ ] Documentation updated, or N/A.
- [ ] Environment variables set in CI, or N/A.

### CODE REVIEW

File: webapp/app/[locale]/tunnel/_components/form.tsx [hunk 1]
```diff
@@ -45,10 +51,18 @@ export const FormContent = function ({
...
+  const hemi = useHemi()
   const t = useTranslations('tunnel-page')
 
   const showToast = useToastIfNotConnectedTo(fromNetworkId)
 
+  const evmTunneling =
+    typeof fromNetworkId === 'number' && typeof toNetworkId === 'number'
+
+  const l1ChainId = fromNetworkId === hemi.id ? toNetworkId : fromNetworkId
+  const l2ChainId = fromNetworkId === l1ChainId ? toNetworkId : fromNetworkId
```
Comment (gabmontes) [inline webapp/app/[locale]/tunnel/_components/form.tsx:64]: Current logic is kind of difficult to read. Consider these options instead:

```suggestion
  const l1ChainId = fromNetworkId === hemi.id ? toNetworkId : fromNetworkId
  const l2ChainId = fromNetworkId !== hemi.id ? toNetworkId : fromNetworkId
```

or

```suggestion
  const [l1ChainId, l2ChainId] = fromNetworkId === hemi.id
    ? [toNetworkId, fromNetworkId] 
    : [fromNetworkId, toNetworkId]
```

Moreover, in which case `l2ChainId` is different from `hemi.id`?

File: webapp/messages/es.json [hunk 2]
```diff
@@ -172,10 +172,27 @@
...
+  "token-custom-drawer": {
+    "add-contract-to-preview": "Añada una dirección de contrato para previsualizar el token...",
+    "add-this-pair": "Agregue este par a la lista de tokens de Hemi para que todos puedan tunelizarlos",
+    "add-token": "Añadir token",
+    "add-token-on-chain": "Añada la dirección del contrato del token en la red de Capa {network}",
+    "address-does-not-match-l2": "No coincide con el par de L1",
+    "heading": "Tunelizar un token ERC-20 existente",
+    "i-am-sure-to-tunnel": "Estoy seguro que quiero tunelizar este token.",
+    "layer-token-address": "Dirección de contrado del token de capa {layer}",
+    "make-a-request-to-add": "Si deseas agregar este par de tokens ERC-20 a la lista de tokens aprobados de Hemi, <link>haz una solicitud</link>.",
```
Comment (gabmontes) [inline webapp/messages/es.json:184]: Do we have a link?

File: webapp/components/tokenSelector/noTokensMatch.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    )
+  }
+
+  const onClick = function () {
+    setCustomTokenAddress(searchText)
+    closeModal()
+  }
+
+  return (
+    <>
```
Comment (ArturDolzan) [inline webapp/components/tokenSelector/noTokensMatch.tsx:47]: I think we can remove this fragment

**Review by ArturDolzan — COMMENTED** [review]
A minor comment otherwise LGTM.

**Pull request merged**